lib/matplotlib/backend_bases.py

```diff
@@ -2552,8 +2564,10 @@ def __init__(self, canvas, num):
         canvas.manager = self  # store a pointer to parent
         self.num = num
 
-        self.key_press_handler_id = self.canvas.mpl_connect('key_press_event',
-                                                            self.key_press)
+        if rcParams['toolbar'] != 'navigation':
+            self.key_press_handler_id = self.canvas.mpl_connect(
+                                                'key_press_event',
+                                                self.key_press)
```

Comment:
This was added to make it possible for users to disable the default keymap
entirely. It is not intended to be internal only, and I even have a
subsection in my book devoted to it. IIRC, the disconnect code handles None
fine.
On Mar 31, 2015 10:33 PM, "Thomas A Caswell" <notifications@github.com>
wrote:

> In lib/matplotlib/backend_bases.py
> <https://github.com/matplotlib/matplotlib/pull/3652#discussion_r27540901>:
>
> > @@ -2552,8 +2564,10 @@ def __init__(self, canvas, num):
> >          canvas.manager = self  # store a pointer to parent
> >          self.num = num
> >
> > -        self.key_press_handler_id = self.canvas.mpl_connect('key_press_event',
> > -                                                            self.key_press)
> > +        if rcParams['toolbar'] != 'navigation':
> > +            self.key_press_handler_id = self.canvas.mpl_connect(
> > +                                                'key_press_event',
> > +                                                self.key_press)
>
> I am torn on this one, I see @fariza <https://github.com/fariza> point
> that this is on it's way out, but I also don't like variable attributes on
> objects (just because you *can* doesn't mean you *should*). On the other
> hand, this is the *only* place in our code base that we reference
> self.key_press_handler_id so it probably does not really matter either
> way.
>
> This would only be used to deregister a call back, does that code cope
> well with None?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/3652/files#r27540901>.
>